### PR TITLE
use faster clamping for Uint8ClampedArray

### PR DESCRIPTION
**Pull request by MaxGraey**

### CODE REVIEW

File: std/assembly/typedarray.ts [hunk 1]
```diff
@@ -12,6 +12,15 @@ import {
   COMPARATOR
 } from "./internal/sort";
 
+@inline
+function clampToByte(value: i32): i32 {
+  if (ASC_OPTIMIZE_LEVEL < 2 || ASC_SHRINK_LEVEL > 0) {
+    return min(max(value, 0), 255);
```
Comment (dcodeIO) [inline std/assembly/typedarray.ts]: Would it also be an option to [optimize](https://graphics.stanford.edu/~seander/bithacks.html#IntegerMinOrMax) `min` and `max` instead? These are currently `select`s.

Comment (MaxGraey) [reply std/assembly/typedarray.ts]: This very special optimization only for `clamp(x, 0, 255)`. I tested bit twidding min/max approach with existing and it doesn't get significant benefit on Firefox, on Chrome this even slowdown on `~15%`.

Comment (MaxGraey) [reply std/assembly/typedarray.ts]: And I tested with different combinations. Compare single `min` and `max`, with compination `max(min(x, a), b)`, with constants and so on. In all cases existing approach faster

Comment (MaxGraey) [reply std/assembly/typedarray.ts]: Removed `@inline`

**Comment by MaxGraey** [comment]
I figure outing new `clamp` approach also smaller than `min/max` so now it uses always except opt == 0

**Comment by MaxGraey** [comment]
Also ready to PR if no more suggestions

**Comment by MaxGraey** [comment]
Also synced with master

**Comment by dcodeIO** [comment]
Suggesting to use the faster version in any case as size difference seems to be minimal (there is no int min/max anyway and these become selects), and putting it into the overloads directly, removing the helper.

**Comment by MaxGraey** [comment]
Ready!

**Comment by dcodeIO** [comment]
Thanks! :)

**Pull request merged**